src/components/Navigator.js

```diff
@@ -3,74 +3,141 @@ import { Subject } from 'rxjs';
+    };
+
+    possibleDirections = getDirectionNames().reduce(reduce, {});
   };
 
   /**
-   * Builds the direction object for one level.
-   *
-   * @param {*[]} elements Elements to look into
-   * @param {number} index index of state at this level
-   * @param {number[]} stateLevel state until level
-   *
-   * @returns {{next, previous}} object with indices of
-   * next and previous indices. See @getIndex
+   * Get motion display names
+   * @returns {string[]} Array of possible motions (left, right, down, up)
+   */
+  let getDirectionNames = () => {
+    return Object.keys(motions);
+  };
+
+  /**
+   * Is direction possible?
+   * @param {string} direction
+   * @returns {boolean}
+   */
+  let isDirectionPossible = (direction) => {
+    return !!possibleDirections[direction];
+  };
+
+  /**
+   * Is state valid?
+   * Tests if state is an array.
+   * @param {*} state
+   * @returns {boolean}
+   */
+  let isValidState = (state) => {
+    return Array.isArray(state);
+  };
+
+  /**
+   * Get level and direction for motion-string
+   * @param {string} direction
+   * @returns {level, direction}
+   */
+  let toLevelAndDirection = (direction) => {
+    return motions[direction];
+  };
+
+  /**
+   * Returns new state from motion
+   * @param {level, direction} motion
+   * @returns {number[]|false}
    */
-  let getDirectionObject = (elements, index, stateLevel) => {
-    return {
-      next: getIndex(elements, index + 1, stateLevel),
-      previous: getIndex(elements, index - 1, stateLevel)
+  let toState = (direction) => {
+    let motion = toLevelAndDirection(direction);
+    return possibleMoves[motion.level] && possibleMoves[motion.level][motion.direction];
+  };
+
+  /**
+   * Pushes new state to subject if state is valid.
+   * @param {*} newState
+   */
+  let pushNewState = (newState) => {
+    if(isValidState(newState)) {
+      subject.next(newState);
     }
   };
 
   /**
-   * Returns the new state based on index or false, if
-   * the state does not exist
-   *
-   * @param {*[]} elements
-   * @param {number} index
-   * @param {number[]} stateLevel state
-   * @returns {boolean|number[]} state list or false
+   * Calculates new state from direction (left/right/up/down)
+   * and if valid, pushes the new state to the subject.
+   * @param {string} direction
    */
-  let getIndex = (elements, index, stateLevel) => {
-    return elements[index] !== undefined && stateLevel.concat([index]) || false;
+  let move = (direction) => {
+    if (!isDirectionPossible(direction)) return;
+    pushNewState(toState(direction));
   };
 
+  /**
+   * If valid, pushes a new state to the subject.
+   * @param {number[]} newState
+   */
+  let go = (newState) => {
+    pushNewState(newState);
+  };
+
+  /**
+   * Get subject to observe navigator changes
+   * @returns {*}
+   */
   let asObservable = () => {
     return subject;
-  }
+  };
 
   return {
-    getDirections,
     asObservable,
-  };
+    setPossibleMoves,
```

Comment:
yep, we need that in `UnveilApp`. Because we need to listen to the `Navigator.asObservable`, I don't create this object on every request, but only once.